codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/ApiKeyAuthDecorator.kt

```diff
@@ -0,0 +1,220 @@
+ */
+class ApiKeyAuthDecorator : ClientCodegenDecorator {
+    override val name: String = "ApiKeyAuth"
+    override val order: Byte = 10
+
+    private fun applies(codegenContext: ClientCodegenContext) =
+        isSupportedApiKeyAuth(codegenContext)
+
+    override fun configCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<ConfigCustomization>,
+    ): List<ConfigCustomization> {
+        return baseCustomizations.letIf(applies(codegenContext)) { customizations ->
+            customizations + ApiKeyConfigCustomization(codegenContext.runtimeConfig)
+        }
+    }
+
+    override fun libRsCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<LibRsCustomization>,
+    ): List<LibRsCustomization> {
+        return baseCustomizations.letIf(applies(codegenContext)) { customizations ->
+            customizations + ApiKeyPubUse(codegenContext.runtimeConfig)
+        }
+    }
+
+    override fun operationCustomizations(
+        codegenContext: ClientCodegenContext,
+        operation: OperationShape,
+        baseCustomizations: List<OperationCustomization>,
+    ): List<OperationCustomization> {
+        if (applies(codegenContext) && hasApiKeyAuthScheme(codegenContext, operation)) {
+            val service = codegenContext.serviceShape
+            val authDefinition: HttpApiKeyAuthTrait = service.expectTrait(HttpApiKeyAuthTrait::class.java)
+            return baseCustomizations + ApiKeyOperationCustomization(codegenContext.runtimeConfig, authDefinition)
+        }
+        return baseCustomizations
+    }
+}
+
+/**
+ * Returns if the service supports the httpApiKeyAuth trait.
+ *
+ * @param codegenContext Codegen context that includes the model and service shape
+ * @return if the httpApiKeyAuth trait is used by the service
+ */
+fun isSupportedApiKeyAuth(codegenContext: ClientCodegenContext): Boolean {
+    return ServiceIndex.of(codegenContext.model).getAuthSchemes(codegenContext.serviceShape).containsKey(HttpApiKeyAuthTrait.ID)
+}
+
+/**
+ * Returns if the service and operation have the httpApiKeyAuthTrait.
+ *
+ * @param codegenContext codegen context that includes the model and service shape
+ * @param operation operation shape
+ * @return if the service and operation have the httpApiKeyAuthTrait
+ */
+fun hasApiKeyAuthScheme(codegenContext: ClientCodegenContext, operation: OperationShape): Boolean {
+    val auth: Map<ShapeId, Trait> = ServiceIndex.of(codegenContext.model).getEffectiveAuthSchemes(codegenContext.serviceShape.getId(), operation.getId())
+    return auth.containsKey(HttpApiKeyAuthTrait.ID) && !operation.hasTrait(OptionalAuthTrait.ID)
+}
+
+private class ApiKeyPubUse(private val runtimeConfig: RuntimeConfig) :
+    LibRsCustomization() {
+    override fun section(section: LibRsSection): Writable = when (section) {
+        is LibRsSection.Body -> writable {
+            rust(
+                "pub use #T;",
```

Comment:
`Credentials` is currently in the crate root, but that will change with [RFC-26](https://github.com/awslabs/smithy-rs/blob/main/design/src/rfcs/rfc0026_client_crate_organization.md) (which I'm currently implementing). Could you make this an `extras` customization and place this re-export in the `crate::config` module instead?

Something like:
```kotlin
    override fun extras(codegenContext: ClientCodegenContext, rustCrate: RustCrate) {
        rustCrate.withModule(ClientRustModule.Config) {
            rust("pub use #T;", apiKey(runtimeConfig))
        }
    }
```
On `ApiKeyAuthDecorator`

Reply:
done